protocol/x/subaccounts/keeper/subaccount.go

```diff
@@ -292,6 +292,13 @@ func (k Keeper) UpdateSubaccounts(
 		perpIdToFundingIndex[perp.Params.Id] = perp.FundingIndex
 	}
 
+	// Check if the updates satisfy the isolated perpetual constraints.
+	success, successPerUpdate, err = k.checkIsolatedSubaccountConstraints(
```

Comment:
Shouldn't we put this in `internalCanUpdateSubaccounts`? I believe we maintain the constraint that `CanUpdateSubaccount` success guarantees `UpdateSubaccounts` success. This will no longer hold true if the input `updates` fails `checkIsolatedSubaccountConstraints`

Reply:
+1 to this comment

Reply:
`CanUpdateSubaccounts` has a different constraint from `UpdateSubaccounts` which is why I decided to pull this function into `UpdateSubaccounts` and not `CanUpdateSubaccounts`.
`CanUpdateSubaccounts` does not require unique subaccounts in updates, but `UpdateSubaccounts` does.
There may not be a use of this specific behavior of `CanUpdateSubaccounts`, I can check, however I preferred changing `UpdateSubaccounts` rather than `CanUpdateSubaccounts` after seeing this difference between the 2 functions.

Reply:
Yes, that's true that it can be added to the orderbook since `AddOrderToOrderbookCollatCheck` calls `CanUpdateSubaccounts`. However this is also a path where having a set of updates with non-unique subaccounts are run through `CanUpdateSubaccounts` but never through `UpdateSubaccounts` and just adding the above check to `CanUpdateSubaccounts` would break a working flow.

Reply:
Removed the unique subaccount check in `checkIsolatedSubaccountConstraints` and moved into `internalCanUpdateSubaccounts`. Specified in comments that the function does not consider constraints across multiple updates.